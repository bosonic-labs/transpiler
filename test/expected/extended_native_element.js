(function () {
    window.SuperButton = document.registerElement('super-button', {
        prototype: Object.create(HTMLButtonElement.prototype, {
            dummy: {
                enumerable: true,
                value: function () {
                    console.log('bar');
                }
            }
        }),
        extends: 'button'
    });
}());