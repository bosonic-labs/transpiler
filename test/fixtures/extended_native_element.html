<element name="super-button" extends="button">
  <script>
    ({
      dummy: function () {
        console.log('bar');
      }
    });
  </script>
</element>