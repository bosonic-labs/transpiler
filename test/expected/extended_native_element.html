<element name="super-button" extends="button">
  <script>
(function () {
    window.SuperButton = document.registerElement('super-button', {
        prototype: Object.create(HTMLButtonElement.prototype, {
            dummy: {
                enumerable: true,
                value: function () {
                    console.log('bar');
                }
            }
        }),
        extends: 'button'
    });
    Object.defineProperty(SuperButton.prototype, '_super', {
        enumerable: false,
        writable: false,
        configurable: false,
        value: HTMLButtonElement.prototype
    });
}());
</script>
</element>